dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIExtension.java

```diff
@@ -59,9 +59,9 @@ public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscover
         AnnotatedType<FaultTolerance> bindingType = beanManager.createAnnotatedType(FaultTolerance.class);
         beforeBeanDiscovery.addInterceptorBinding(bindingType);
         AnnotatedType<FaultToleranceInterceptor> interceptorType = beanManager.createAnnotatedType(FaultToleranceInterceptor.class);
-        beforeBeanDiscovery.addAnnotatedType(interceptorType);
+        beforeBeanDiscovery.addAnnotatedType(interceptorType, (interceptorType != null) ? interceptorType.getClass().getName() + ": " + interceptorType.getClass().hashCode() : "");
```

Comment:
Is the null check actually needed here?

Reply:
I discussed this with Tom.  We reckon, that with the current system it probably is not - but we could not state that it definately was not or that this would not change. So I decided to be safe - it is one machine instruction. 
The other 'odd' thing you might be thinking is 'why call the weld method if the value IS null' and the reason is that I want weld to react and get a weld error (etc.)  rather than a NPE in a getClass().